webui/react/src/components/ExperimentRetainLogsModal.tsx

```diff
@@ -0,0 +1,142 @@
+import Checkbox, { CheckboxChangeEvent } from 'hew/Checkbox';
+import Form from 'hew/Form';
+import InputNumber from 'hew/InputNumber';
+import { Modal } from 'hew/Modal';
+import { useToast } from 'hew/Toast';
+// import Select, { Option } from 'hew/Select';
+// import Spinner from 'hew/Spinner';
+// import { useToast } from 'hew/Toast';
+// import { Label } from 'hew/Typography';
+import React, { useId, useState } from 'react';
+
+// import usePermissions from 'hooks/usePermissions';
+import { changeExperimentLogRetention } from 'services/api';
+import { V1BulkExperimentFilters } from 'services/api-ts-sdk';
+import handleError from 'utils/error';
+import { pluralizer } from 'utils/string';
+
+const FORM_ID = 'retain-experiment-logs-form';
+
+type FormInputs = {
+  numDays: number;
+};
+
+interface Props {
+  excludedExperimentIds?: Set<number>;
+  experimentIds: number[];
+  filters?: V1BulkExperimentFilters;
+  onSubmit?: (successfulIds?: number[]) => void;
+}
+
+const ExperimentRetainLogsModalComponent: React.FC<Props> = ({
+  excludedExperimentIds,
+  experimentIds,
+  filters,
+  onSubmit,
+}: Props) => {
+  const idPrefix = useId();
+  const { openToast } = useToast();
+  // const [disabled, setDisabled] = useState<boolean>(true);
+  const [checked, setChecked] = useState<boolean>(false);
+  const [form] = Form.useForm<FormInputs>();
+
+  const handleCheckBoxChange = (event: CheckboxChangeEvent) => {
+    const isChecked = event.target.checked;
+    setChecked(isChecked);
+    if (isChecked) {
+      form.setFieldValue('numDays', -1);
+    }
+  };
+
+  const handleSubmit = async () => {
+    const values = await form.validateFields();
+    const numberDays = values.numDays;
+    if (excludedExperimentIds?.size) {
```

Comment:
its good to use a border number like `> 0` for readability

Reply:
done!